src/test/java/com/iopipe/__DoEventInfoPlugin__.java

```diff
@@ -0,0 +1,336 @@
+import com.fasterxml.jackson.datatype.joda.JodaModule;
+import com.iopipe.http.RemoteRequest;
+import com.iopipe.http.RemoteResult;
+import com.iopipe.IOpipeMeasurement;
+import com.iopipe.plugin.eventinfo.EventInfoDecoder;
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.IOException;
+import java.util.function.Supplier;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.regex.Pattern;
+import java.util.Set;
+import javax.json.JsonArray;
+import javax.json.JsonNumber;
+import javax.json.JsonObject;
+import javax.json.JsonString;
+import javax.json.JsonValue;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.LogManager;
+
+/**
+ * This tests that the event info plugin detects the input event sources for
+ * object correctly.
+ *
+ * @since 2018/04/16
+ */
+class __DoEventInfoPlugin__
+	extends Single
+{
+	/** Logging. */
+	private static final Logger _LOGGER =
+		LogManager.getLogger(__DoEventInfoPlugin__.class);
+	
+	/** Sent with no exception? */
+	protected final BooleanValue noerror =
+		new BooleanValue("noerror");
+		
+	/** Got a result from the server okay? */
+	protected final BooleanValue remoterecvokay =
+		new BooleanValue("remoterecvokay");
+	
+	/** Got the event type? */
+	protected final BooleanValue typegot =
+		new BooleanValue("typegot");
+	
+	/** Metrics were found */
+	protected final BooleanValue gotmetrics =
+		new BooleanValue("gotmetrics");
+	
+	/** The decoder that is used. */
+	protected final EventInfoDecoder decoder;
+	
+	/**
+	 * Constructs the test.
+	 *
+	 * @param __e The owning engine.
+	 * @param __input The input.
+	 * @param __decoder The decoder being used.
+	 * @throws NullPointerException On null arguments.
+	 * @since 2018/04/16
+	 */
+	__DoEventInfoPlugin__(Engine __e, Supplier<Object> __input,
+		EventInfoDecoder __decoder)
+		throws NullPointerException
+	{
+		super(__e, "eventinfo-" + __decoder.eventType(), __input);
+		
+		if (__decoder == null)
+			throw new NullPointerException();
+		
+		this.decoder = __decoder;
+	}
+	
+	/**
+	 * {@inheritDoc}
+	 * @since 2018/04/16
+	 */
+	@Override
+	public void end()
+	{
+		super.assertTrue(this.remoterecvokay);
+		super.assertTrue(this.noerror);
+		
+		// Must have read all metrics and the event type
+		super.assertTrue(this.typegot);
+		super.assertTrue(this.gotmetrics);
+	}
+	
+	/**
+	 * {@inheritDoc}
+	 * @since 2018/04/18
+	 */
+	@Override
+	public void modifyConfig(IOpipeConfigurationBuilder __cb)
+		throws NullPointerException
+	{
+		if (__cb == null)
+			throw new NullPointerException();
+		
+		__cb.setPluginEnabled("event-info", true);
+	}
+	
+	/**
+	 * {@inheritDoc}
+	 * @since 2018/04/16
+	 */
+	@Override
+	public void remoteRequest(WrappedRequest __r)
+	{
+		EventInfoDecoder decoder = this.decoder;
+		Map<String, JsonValue> expand = __Utils__.expandObject(__r.request);
+		
+		// It is invalid if there is an error
```

Comment:
It is invalid if there is an error? What does this mean here?

Reply:
This check here is for when IOpipe reports when the method that was invoked threw an exception, this just checks that no error was actually thrown.